azurerm/internal/services/containers/kubernetes_addons.go

```diff
@@ -154,6 +157,38 @@ func schemaKubernetesAddOnProfiles() *schema.Schema {
 						},
 					},
 				},
+
+				"ingress_application_gateway": {
+					Type:     schema.TypeList,
+					MaxItems: 1,
+					Optional: true,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"enabled": {
+								Type:     schema.TypeBool,
+								Required: true,
+							},
+							"application_gateway_id": {
+								Type:         schema.TypeString,
+								Optional:     true,
+								ValidateFunc: applicationGatewayValidate.ApplicationGatewayID,
+							},
+							"effective_application_gateway_id": {
+								Type:     schema.TypeString,
+								Computed: true,
+							},
+							"subnet_cidr": {
```

Comment:
Assuming the `gateway_id`, `subnet_cidr` and `subnet_id` are exclusively used, then shall we add the `ConflictsWith` constraint for each of these properties?